data-processing/common/parse_equation.py

```diff
@@ -64,23 +66,54 @@ def get_symbols(mathml: str) -> List[Symbol]:
             if index is not None:
                 characters.append(index)
 
-        # Clean the MathML of our annotations. Do this cleaning on a clone of the node, so that the
-        # annotations are available for processing later nodes.
-        node_clone = copy.copy(symbol_node)
-        _remove_s2_annotations(node_clone)
-        for descendant in node_clone.descendants:
-            _remove_s2_annotations(descendant)
+        # Group consecutive character symbols only if they are a child of the mrow tag.
+        # The .has_attr('s2:end') is necessary as not every symbol is annotated by katex
+        if symbol_node.has_attr('s2:end') and symbol_node.parent.name in VALID_PARENT_TAG and symbol_node.name in CHARACTER_TAGS:
```

Comment:
nit pick: wrap line at 100 characters. Are you using black and pylint?

Reply:
Ah, I never shared with you the settings.json for the project. Try loading these options into your settings.json in VSCode. It should catch the small formatting and typing things.

```json
{
  "python.linting.enabled": true,
  "python.linting.pylintEnabled": true,
  "python.testing.pytestArgs": [
    "tests"
  ],
  "python.linting.pylintArgs": [
    "--extension-pkg-whitelist=cv2",
    "--disable=missing-docstring,len-as-condition,fixme,invalid-name,anomalous-backslash-in-string,line-too-long,bad-continuation,no-self-use,too-many-locals,too-few-public-methods,ungrouped-imports,pointless-string-statement,too-many-arguments"
  ],
  "python.testing.unittestEnabled": false,
  "python.testing.nosetestsEnabled": false,
  "python.testing.pytestEnabled": true,
  "python.linting.mypyEnabled": true,
  "python.linting.mypyArgs": [
    "--ignore-missing-imports",
    "--follow-imports=silent",
    "--show-column-numbers",
    "--strict"
  ]
}
```

Reply:
I just opened issue #64 for myself to add the settings.json file in the next couple of days. In the meantime, go ahead and incorporate this into your project directory!